### DISCUSSION TITLE
Accessing Inertia via context api?

### DISCUSSION THREAD

**Discussion by beeglebug** [inertiajs/inertia#180] [Help (React)]
I'm trying to split my javascript code down into multiple packages to share components between several applications. 

When i try to use`Inertia.visit` inside one of these packages, it fails with `this.resolveComponent is not defined`. From looking into the source, I think this is something to do with having multiple instances of Inertia, and only the one created inside `App` has called init and set itself up correctly?

Is there any way this root copy of Inertia could be shared with all child elements, possibly via the context api? Something like:

```javascript
const intertia = useInertia() // returns the root initialised inertia instance
const onClick = () => inertia.visit('/')
```

I'm thinking closer to how a lot of other react navigation libraries work, where if you attempt to use a link component outside of a router, you get an error about not having the correct context above it.

**Comment by sebastiandedeyne** [comment]
I generally think this would make sense. 

However, Inertia itself is a singleton, there's always only one instance, so from that perspective a setup like this would look odd.

`this.resolveComponent is not defined` sounds like something we should solve though, although it's requires a more high level discussion that just for `inertia-react`, since other libraries will have the same problem when calling Inertia without setting it up (e.g. for testing purposes).

**Comment by reinink** [comment]
Gents, help me understand why you can't just call `Inertia.visit()` from anywhere in your applications? Inertia is just a singleton, so assuming you have a base package (your app) that sets up Inertia, you should be able to just import it and access the current instance. What am I missing here?
